server/src/main/java/com/google/coffeehouse/servlets/GetProfileServlet.java

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.coffeehouse.servlets;
+
+import com.google.coffeehouse.common.Person;
+import com.google.coffeehouse.storagehandler.StorageHandlerApi;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.Map;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * Servlet to get a {@link Person} from Http GET Request Body (in JSON format)
+ * that exists in the database through a call to the Storage Handler API,
+ * and return it in JSON format.
+ */
+@WebServlet("/api/get-profile")
+public class GetProfileServlet extends HttpServlet {
+
+  /** 
+   * The error string sent by the response object in doGet when the body of the 
+   * GET request cannot be used to fetch a {@link Person} for any reason.
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+
+  /** The logged error string when an error parsing the body of the GET request is encountered */
+  public static final String LOG_BODY_ERROR_MESSAGE = 
+      "Body unable to be parsed in GetProfileServlet: ";
+  private static final Gson gson = new Gson();
+  private StorageHandlerApi storageHandler = new StorageHandlerApi();
```

Comment:
Consider making this static and final and moving instantiation into the default constructor.

Reply:
I'm having issues compiling if I set it to `private static final StorageHandlerApi storageHandler;` and then instantiate it within the default constructor, since I can't assign a value to a final variable. It compiles if it is just static, however, I'm not sure if that's what you would want me to do. Would it be possible to get further clarification on this comment?

Reply:
Sorry, just `final` would be appropriate here. 
So I would recommend having the line `private final StorageHandlerApi storageHandler;` here and initialize it in each constructor.

Reply:
Got it!